com.unity.multiplayer.mlapi/Prototyping/NetworkTransform.cs

```diff
@@ -11,364 +11,217 @@ namespace MLAPI.Prototyping
-                    }
-                    else
-                    {
-                        SubmitTransformServerRpc(transform.position, transform.rotation.eulerAngles);
-                    }
+                    v.Value = initialValue;
                 }
+                oldVal = initialValue;
             }
-            else
-            {
-                //If we are server and interpolation is turned on for server OR we are not server and interpolation is turned on
-                if ((IsServer && InterpolateServer && InterpolatePosition) || (!IsServer && InterpolatePosition))
-                {
-                    if (Vector3.Distance(transform.position, m_LerpEndPos) > SnapDistance)
-                    {
-                        //Snap, set T to 1 (100% of the lerp)
-                        m_LerpTime = 1f;
-                    }
-
-                    float sendDelay = (IsServer || !EnableRange || !AssumeSyncedSends || NetworkManager.ConnectedClients[NetworkManager.LocalClientId].PlayerObject == null) ? (1f / FixedSendsPerSecond) : GetTimeForLerp(transform.position, NetworkManager.ConnectedClients[NetworkManager.LocalClientId].PlayerObject.transform.position);
-                    m_LerpTime += Time.unscaledDeltaTime / sendDelay;
 
-                    if (ExtrapolatePosition && Time.unscaledTime - m_LastReceiveTime < sendDelay * MaxSendsToExtrapolate)
-                    {
-                        transform.position = Vector3.LerpUnclamped(m_LerpStartPos, m_LerpEndPos, m_LerpTime);
-                    }
-                    else
-                    {
-                        transform.position = Vector3.Lerp(m_LerpStartPos, m_LerpEndPos, m_LerpTime);
-                    }
+            SetupVar(m_NetworkPosition, m_Transform.position, ref m_OldPosition);
+            SetupVar(m_NetworkRotation, m_Transform.rotation, ref m_OldRotation);
+            SetupVar(m_NetworkWorldScale, m_Transform.lossyScale, ref m_OldScale);
 
-                    if (ExtrapolatePosition && Time.unscaledTime - m_LastReceiveTime < sendDelay * MaxSendsToExtrapolate)
-                    {
-                        transform.rotation = Quaternion.SlerpUnclamped(m_LerpStartRot, m_LerpEndRot, m_LerpTime);
-                    }
-                    else
-                    {
-                        transform.rotation = Quaternion.Slerp(m_LerpStartRot, m_LerpEndRot, m_LerpTime);
-                    }
-                }
-            }
-
-            if (IsServer && EnableRange && EnableNonProvokedResendChecks)
+            m_PositionChangedDelegate = GetOnValueChangedDelegate<Vector3>(current =>
             {
-                CheckForMissedSends();
-            }
-        }
+                transform.position = current;
+                m_OldPosition = current;
+            });
+            m_NetworkPosition.OnValueChanged += m_PositionChangedDelegate;
```

Comment:
Just wondering why we are doing this in such a complicated way. Why are we storing the delegate as a member. Why are we manually creating the delegate instead of just defining functions and using implicit conversation?

Reply:
I'm storing the delegate to be able to unsubscribe in the OnDestroy
How would you use implicit conversion to differentiate between position and scale? They are both Vector3

Reply:
Oh I see this is for the custom `GetOnValueChangedDelegate` to ensure that the delegate is not called twice with client authority.